programming-extensions/programming-extension-vfsprovider/src/main/java/org/objectweb/proactive/extensions/dataspaces/vfs/VFSFactory.java

```diff
@@ -244,8 +278,38 @@ private static FilesCache getCacheType() {
         return cache;
     }
 
-    private static FileSystemOptions createDefaultFileSystemOptions() throws FileSystemException {
+    private static synchronized FileSystemOptions createDefaultFileSystemOptions(UserCredentials credentials)
+            throws FileSystemException {
+
         final FileSystemOptions options = new FileSystemOptions();
+        if (logger.isTraceEnabled()) {
+            logger.trace("CREDENTIALS = " + credentials);
+        }
+        if (credentials != null && !credentials.isEmpty()) {
+            if (Strings.isNullOrEmpty(credentials.getPassword()) &&
+                (credentials.getPrivateKey() != null && credentials.getPrivateKey().length > 0)) {
+                try {
+                    // unfortunately, the current vfs2 sftp api allows only to provide an identify as a file
+                    File tempFile = File.createTempFile("sftp", "");
```

Comment:
Is there a way to clean the tempFile after the authentication?  To avoid exposing users private key.